install.sh

```diff
@@ -0,0 +1,212 @@
+    echo '   commandline                   Runs command line version of LanguageTool'
+    echo '   server                        Runs server version of LanguageTool'
+    echo
+    echo 'Packages(only if -b is specified):'
+    echo '   standalone                  Installs standalone package'
+    echo '   wikipedia                   Installs Wikipedia package'
+    echo '   office-extension            Installs the LibreOffice/OpenOffice extension package'
+    echo
+    echo 'Submit a GitHub issue if you are encountering problems or want to suggest new features'
+    echo
+    exit 1
+}
+
+install() {
+    echo "Removing any old copy of LanguageTools Stable in this directory"
+    rm LanguageTool-stable.zip &>/dev/null
+
+    echo "Installing LanguageTools Stable"
+    version=stable
+    RELEASE_URL="https://languagetool.org/download/LanguageTool-stable.zip"
+    curl -l $RELEASE_URL -o LanguageTool-stable.zip
+    DIR=$(unzip LanguageTool-stable.zip | grep -m1 'creating:' | cut -d' ' -f5- )
+    rm -r $DIR &>/dev/null
+    RELEASE=${DIR%/}
+
+    # Getting rid of any old folders with the same name
+    rm -r $RELEASE-$version
+
+    echo "Unzipping"
+    unzip -u LanguageTool-stable.zip
+
+    mv $RELEASE "$RELEASE-$version"
+    echo "Cleaning up"
+    rm LanguageTool-stable.zip
+}
+
+build () {
+    # Cloning from GitHub
+    git clone --depth 1 https://github.com/languagetool-org/languagetool.git
+    cd languagetool || exit 2
+
+    # Seeing if maven is installed, and installing it if not
+    detect_maven
+
+    mvn clean test
+    if [ "$specifypackage" = YES ]; then
+        ./build.sh languagetool-$package package
+    else
+        ./build.sh languagetool-standalone package
+        ./build.sh languagetool-wikipedia package -DskipTests
+        ./build.sh languagetool-office-extension package -DskipTests
+    fi
+}
+
+install_maven() {
+    echo "Installing maven . . ."
+    if [[ "$OSTYPE" == "linux-gnu" ]]; then
+        sudo apt-get update -y
+        sudo apt-get install maven
+
+    elif [[ "$OSTYPE" == "darwin"* ]]; then
+        if ! [ -x "$(brew -v)" ]; then
+               /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+        fi
+        brew update
+        brew install maven
+
+    else
+            echo "Error: java is not installed and operating system detection error"
+            echo "   OS type not supported!"
+            echo "   Please install maven yourself or override automatic OS detection with -o <OS> See help for more details."
+    fi
+}
+
+install_java() {
+    echo "Java is not installed"
+    echo "Installing java . . ."
+
+    if [[ "$OSTYPE" == "linux-gnu" ]]; then
+        sudo apt-get update
+        sudo apt-get install java -y
+
+    elif [[ "$OSTYPE" == "darwin"* ]]; then
+        if ! [ -x "$(brew -v)" ]; then
+               /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
```

Comment:
Tried this script moments ago on Mint 18.3.
It ends up not detecting JavaJDK when -d is used.
Needs: `sudo apt-get install openjdk-8-jdk`.
Also #JAVA_HOME was not defined, but I am not sure if this is solved after openjdk-8-jdk installation.

Still not thoroughly tested but, when ready, it will be a time-saver.